app/plugin_api.go

```diff
@@ -914,3 +914,59 @@ func (api *PluginAPI) PluginHTTP(request *http.Request) *http.Response {
 	api.app.ServeInterPluginRequest(responseTransfer, request, api.id, destinationPluginId)
 	return responseTransfer.GenerateResponse()
 }
+
+func (api *PluginAPI) CreateSlashCommand(cmd *model.Command) (*model.Command, *model.AppError) {
+	cmd.CreatorId = ""
+	cmd.PluginId = api.id
+
+	cmd, appErr := api.app.CreateCommand(cmd)
+
+	return cmd, appErr
+}
+
+func (api *PluginAPI) ListSlashCommands(teamID string, customOnly bool) ([]*model.Command, *model.AppError) {
+	var commands []*model.Command
+	var appErr *model.AppError
+
+	if customOnly {
+		commands, appErr = api.app.ListTeamCommands(teamID)
+	} else {
+		commands, appErr = api.app.ListAllCommands(teamID, api.app.T)
+	}
+	return commands, appErr
+}
+
+func (api *PluginAPI) GetSlashCommand(commandID string) (*model.Command, *model.AppError) {
+	return api.app.GetCommand(commandID)
+}
+
+func (api *PluginAPI) UpdateSlashCommand(commandID string, cmd *model.Command) (*model.Command, *model.AppError) {
+	oldCmd, appErr := api.GetSlashCommand(commandID)
+	if appErr != nil {
+		return nil, appErr
+	}
+
+	if cmd.TeamId != oldCmd.TeamId {
+		return nil, model.NewAppError("UpdateSlashCommand", "api.command.team_mismatch.app_error", nil, "plugin_id="+api.id, http.StatusBadRequest)
+	}
+
+	return api.app.UpdateCommand(oldCmd, cmd)
+}
+
+func (api *PluginAPI) MoveSlashCommand(commandID string, newTeamID string) *model.AppError {
```

Comment:
I know this was discussed elsewhere, but should we take the opportunity to simplify the API and just support moving directly in `UpdateSlashCommand`?

Reply:
Talked to Joram, his message:

Two concerns I can think of:

1. Changing this will change the behaviour of the API (both REST and plugin) and could potentially break consumers relying on that behaviour (e.g. someone is using the API to update their commands but is not providing a team ID)
1. The current REST API for update command doesn't do any permission checks on the new team ID so we'd need to do that (not a concern for plugins)

Reply:
Thanks, for the follow-up, @cpoile. I think we've already long since diverged from the REST API proper, so I definitely wouldn't propose touching the REST API there. My chief focus is how we can make the plugin API "better" and drop the legacy behaviour that was needed to support the REST API.

Along those lines, should we require a `teamID` when calling the plugin's `UpdateCommand`, and transparently support moving it between teams when it changes?

Reply:
Sorry, I totally misunderstood what you meant. 🤦  That makes total sense.
Now I'm transparently moving between teams if `updatedCommand` has a new `teamId`, if it has no `teamId` then we'll use the `oldCmd`'s teamId. The comments are updated to reflect this too.